lib/iota.js

```diff
@@ -24,14 +24,15 @@ IOTA.prototype.setSettings = function(settings) {
     this.provider = settings.provider || this.host.replace(/\/$/, '') + ":" + this.port;
     this.sandbox = settings.sandbox || false;
     this.token = settings.token || false;
-
+    this.username = settings.username || false;
+    this.password = settings.password || false;
     if (this.sandbox) {
         // remove backslash character
         this.sandbox = this.provider.replace(/\/$/, '');
         this.provider = this.sandbox + '/commands';
     }
 
-    this._makeRequest = new makeRequest(this.provider, this.token);
+    this._makeRequest = new makeRequest({provider: this.provider, token: this.token, username: this.username, password: this.password});
```

Comment:
Could you pass `username` & `password` as separate optional arguments instead of object?
For users calling `iota._makeRequest` this is a breaking change.